lib/view_component/compiler.rb

```diff
@@ -5,70 +5,90 @@ class Compiler
     # Lock required to be obtained before compiling the component
     attr_reader :__vc_compiler_lock
 
+    # Compiler mode. Can be either:
+    # * blocking (default in Rails development and test mode)
+    # * non_blocking (default in Rails production mode)
+    attr_accessor :mode
+
     def initialize(component_class)
       @component_class = component_class
       @__vc_compiler_lock = Monitor.new
+      @mode = defined?(Rails.root) && (Rails.env.development? || Rails.env.test?) ? :blocking : :non_blocking
```

Comment:
Why `defined?(Rails.root)` specifically? Is the intent here to check whether we're running within the context of a full Rails app, or to check that Rails is installed at all?

Reply:
I'm inclined to agree, but it looks to me like we're using that same check a lot elsewhere, so I guess it makes sense to keep it around here.

Reply:
The other way around, but yes! Good idea